### DISCUSSION TITLE
prefer-const with array destructuring

### DISCUSSION THREAD

**Discussion by yukulele** [eslint/eslint#17087] [Q&A] [answered]
in this example I use array destructuring to get `key` and `value` from `array.entries()`
```javascript
const array = [10, 20, 30]
let total = 0
for (let [index, value] of array.entries()) {
  value /= 2
  total = value + index
}
```
I use `let` because `value` is modified but I get a `prefer-const` error for `index`.

Shouldn't using `let` be considered acceptable in such cases?

**Comment by aladdin-add** [comment, marked as answer]
there is an option: `destructuring: "all"` to achieve this:

https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IHByZWZlci1jb25zdDogWzIsIHtkZXN0cnVjdHVyaW5nOiBcImFsbFwifV0qL1xuY29uc3QgYXJyYXkgPSBbMTAsIDIwLCAzMF1cbmxldCB0b3RhbCA9IDBcbmZvciAobGV0IFtpbmRleCwgdmFsdWVdIG9mIGFycmF5LmVudHJpZXMoKSkge1xuICB2YWx1ZSAvPSAyXG4gIHRvdGFsID0gdmFsdWUgKyBpbmRleFxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsicHJlZmVyLWNvbnN0IjpbImVycm9yIl19LCJlbnYiOnt9fX0=

**Reply by yukulele** [reply]
thanks! https://eslint.org/docs/latest/rules/prefer-const#destructuring
